packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -2,96 +2,87 @@ import React from 'react';
+   * The variant of notification to be rendered. This renders the dialog with the corresponding header and icon to the
+   * variant concept.
+   * Use one of `alert`, `error`, `warning`, `info`, or `custom`.
    */
-  header: PropTypes.string,
+  variant: PropTypes.oneOf([
+    variants.ALERT,
+    variants.ERROR,
+    variants.WARNING,
+    variants.INFO,
+    variants.CUSTOM,
+  ]).isRequired,
   /**
-   * Title of the notification-dialog.
+   * The title to describe the high-level overview of why the notification-dialog is being displayed to the user.
    */
   title: PropTypes.string,
   /**
-   * **Deprecated**, Message of the notification-dialog. Resolves to `startMessage`.
-   */
-  message: PropTypes.string,
-  /**
-   * Message to be displayed at the start of the notification dialog body.
+   * The text to provide more detail or defined terminology to be displayed at the start of the notification dialog body.
    */
   startMessage: PropTypes.string,
   /**
-   * Message to be displayed at the end of the notification dialog body.
+   * The text to provide more detail or defined terminology to be displayed at the end of the notification dialog body.
    */
   endMessage: PropTypes.string,
   /**
-   *  Non-plain text content to be inserted after `startMessage` and/or before `endMessage`
+   *  The content to be inserted after `startMessage` and/or before `endMessage` to provide more details to the user in the dialog body.
    */
   content: PropTypes.node,
   /**
-   * **Deprecated**, The Action of the primary button. Resolves to `acceptAction`
-   */
-  primaryAction: PropTypes.shape({
-    text: PropTypes.string,
-    onClick: PropTypes.func,
-  }),
-  /**
-   * The Action of the accept button.
+   * The button text and onclick values of the accept button.
    */
   acceptAction: PropTypes.shape({
     text: PropTypes.string,
     onClick: PropTypes.func,
   }),
   /**
-   * **Deprecated**, The Action of the secondary button. Resolves to `rejectAction`.
-   */
-  secondaryAction: PropTypes.shape({
-    text: PropTypes.string,
-    onClick: PropTypes.func,
-  }),
-  /**
-   * The Action of the reject button.
+   * The button text and onclick values of the reject button.
    */
   rejectAction: PropTypes.shape({
     text: PropTypes.string,
     onClick: PropTypes.func,
   }),
   /**
-   * The variant of notification to be rendered.
-   * Use one of `alert`, `error`, `warning`, `info`, `success`, `custom`.
-   */
-  variant: PropTypes.oneOf([
-    variants.ALERT,
-    variants.ERROR,
-    variants.WARNING,
-    variants.INFO,
-    variants.SUCCESS,
-    variants.CUSTOM,
-  ]),
-  /**
-   * The icon to be used for a notification of type custom. This will not be used for any other notification types.
+   * The pieces to populate a notification-dialog when `variant="custom"`.
    */
-  customIcon: PropTypes.element,
+  custom: PropTypes.shape({
+    /**
+     * The signal word to use in the notification-dialog.
```

Comment:
Could there be a little more explaination of the purpose of a `signal word`?

Reply:
Yeah, the zeplin callouts define it as:
> The keyword for a notification dialog is based on the notification that is being presented to the user (Alert, Warning, Information, etc). These are locked down pairings with specific icons

Based on this, I could update the description to be a bit more description... is this better?
> The keyword used to represent / emphasis the intention of dialog message that that is being shown to the user.

Reply:
I think this might still be a little confusing at first, but that description is easier to understand. Traditionally I think we call these `heading/title`, but I assume we want to drift away from that pattern. `keyword` doesn't sound all that bad. With a clear description consumers should be able to figure `signalWord` out.

Reply:
Neil wants to get away from the terminology heading/header since in other components this implies one can drop in their own header component. 

updated: https://github.com/cerner/terra-framework/pull/1105/commits/022584400b4aaafcf4ad53ff1ed2e2b2eaec08a5#diff-cb62dc7ce0197bc569a9952450bb9fe5R68

Reply:
When we update Alert/Notification-Banners and Status Views, we'll also update to use the SignalWord propname as the go forward - since it has key implications with regulatory + ANSI safety specification standards. I'll see if the safety documentation I have includes anything different for a definition to add to the description Emily just updated.

Reply:
@StephenEsser - added a related comment with regard to `title` as well: https://github.com/cerner/terra-framework/pull/1105/files#r424696268